lib/change_stream_simulator.js

```diff
@@ -0,0 +1,244 @@
+var Timestamp = require('mongodb-core').BSON.Timestamp;
+var MongoNetworkError = require('mongodb-core').MongoNetworkError;
+var supportedStages =  ['$match', '$project', '$addFields', '$replaceRoot', '$redact'];
+
+var createChangeStreamCursorSimulation = function (serverConfig, namespace, pipeline, options) {
+  var DB = require('./db');
```

Comment:
why require `DB` inside the simulator? Is there something dynamic about this? Otherwise, I would just require it above.